contracts/token/ERC20/extensions/ERC1363.sol

```diff
@@ -0,0 +1,151 @@
+
+/**
+ * @title ERC1363
+ * @dev Implementation of the ERC1363 interface.
+ */
+abstract contract ERC1363 is ERC20, ERC165, IERC1363 {
+    /**
+     * @dev Indicates a failure with the token `receiver`. Used in transfers.
+     * @param receiver Address to which tokens are being transferred.
+     */
+    error ERC1363InvalidReceiver(address receiver);
+
+    /**
+     * @dev Indicates a failure with the token `spender`. Used in approvals.
+     * @param spender Address that may be allowed to operate on tokens without being their owner.
+     */
+    error ERC1363InvalidSpender(address spender);
+
+    /**
+     * @inheritdoc IERC165
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165, IERC165) returns (bool) {
+        return interfaceId == type(IERC1363).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @inheritdoc IERC1363
+     */
+    function transferAndCall(address to, uint256 value) public virtual returns (bool) {
+        return transferAndCall(to, value, "");
+    }
+
+    /**
+     * @inheritdoc IERC1363
+     */
+    function transferAndCall(address to, uint256 value, bytes memory data) public virtual returns (bool) {
+        transfer(to, value);
+        _checkOnTransferReceived(_msgSender(), to, value, data);
+        return true;
+    }
+
+    /**
+     * @inheritdoc IERC1363
+     */
+    function transferFromAndCall(address from, address to, uint256 value) public virtual returns (bool) {
+        return transferFromAndCall(from, to, value, "");
+    }
+
+    /**
+     * @inheritdoc IERC1363
+     */
+    function transferFromAndCall(
+        address from,
+        address to,
+        uint256 value,
+        bytes memory data
+    ) public virtual returns (bool) {
+        transferFrom(from, to, value);
+        _checkOnTransferReceived(from, to, value, data);
+        return true;
+    }
+
+    /**
+     * @inheritdoc IERC1363
+     */
+    function approveAndCall(address spender, uint256 value) public virtual returns (bool) {
+        return approveAndCall(spender, value, "");
+    }
+
+    /**
+     * @inheritdoc IERC1363
+     */
+    function approveAndCall(address spender, uint256 value, bytes memory data) public virtual returns (bool) {
+        approve(spender, value);
+        _checkOnApprovalReceived(spender, value, data);
+        return true;
+    }
+
+    /**
+     * @dev Private function to invoke `onTransferReceived` on a target address.
+     * This will revert if the target doesn't implement the `IERC1363Receiver` interface or
+     * if the target doesn't accept the token transfer or
+     * if the target address is not a contract.
+     *
+     * @param from Address representing the previous owner of the given token amount.
+     * @param to Target address that will receive the tokens.
+     * @param value The amount of tokens to be transferred.
+     * @param data Optional data to send along with the call.
```

Comment:
These don't show up in the docs and regardless, I don't think they're needed because the function is private.

I'd just remove them

```suggestion
```